plugins/uefi/fu-plugin-uefi.c

```diff
@@ -135,6 +137,202 @@ fu_plugin_get_results (FuPlugin *plugin, FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_plugin_uefi_update_resource (fwup_resource *re,
+				guint64 hardware_instance,
+				GBytes *blob,
+				GError **error)
+{
+	int rc;
+	rc = fwup_set_up_update_with_buf (re, hardware_instance,
+					  g_bytes_get_data (blob, NULL),
+					  g_bytes_get_size (blob));
+	if (rc < 0) {
+		g_autoptr(GString) str = g_string_new (NULL);
+		rc = 1;
+		for (int i = 0; rc > 0; i++) {
+			char *filename = NULL;
+			char *function = NULL;
+			char *message = NULL;
+			int line = 0;
+			int err = 0;
+
+			rc = efi_error_get (i, &filename, &function, &line,
+					    &message, &err);
+			if (rc <= 0)
+				break;
+			g_string_append_printf (str, "{error #%d} %s:%d %s(): %s: %s, ",
+						i, filename, line, function,
+						message, strerror (err));
+		}
+		if (str->len > 2)
+			g_string_truncate (str, str->len - 2);
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "UEFI firmware update failed: %s",
+			     str->str);
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static GBytes *
+fu_plugin_uefi_get_splash_data (guint width, guint height, GError **error)
```

Comment:
Why not make this `GBoolean `and pass a pointer to the `GBytes `to fill?

Reply:
It would have to be a `GBytes **` and I'm not sure it's much clearer than returning the data or NULL itself.

Reply:
Yeah you're right, it's not.